src/main/java/edu/harvard/iq/dataverse/api/Datasets.java

```diff
@@ -1020,7 +1020,7 @@ public Response publishDataset(@PathParam("id") String id, @QueryParam("type") S
             PublishDatasetResult res = execCommand(new PublishDatasetCommand(ds,
                         createDataverseRequest(user),
                     isMinor));
-            return res.isCompleted() ? ok(json(res.getDataset())) : accepted(json(res.getDataset()));
+            return res.isWorkflow() ? accepted(json(res.getDataset())) : ok(json(res.getDataset()));
```

Comment:
does this result in a 200 when the dataset is still inprogress/publishing not yet finalized? Seems like 202 is the right code for that (as it was) and the test should be watching for a 202?

Reply:
With the async publication in progress it returns a 200; the presence of the "finalizing" lock is the indication of it being in progress. 
To me it seemed like the 202 return code was specifically added for the workflow situation; and there may be some third party code that relies on it in that context. When the async publication was added we just used that mechanism that was there for the workflow.  But using the same code for this scenario seemed wrong. 
And most importantly, yes, this allowed me not to have to modify much more RestAssured code.

Reply:
I understand the practicality and don't think it's a big issue in terms of any effect, but I think 202 is the right code for anything asynchronous that could fail after you get your response. If its not worth addressing now, I'd suggest a ToDo/note in the code.

Reply:
So maybe the same code 202, but something in the body to differentiate from the "workflow" state. As it was implemented, it was returning the code 202 and `{"status":"WORKFLOW...}` in both cases - which was wrong.
It feels like this warrants its own issue - I'll open it.